### ISSUE TITLE
Enquiring about missing step in TSNE analysis

### ISSUE THREAD

**Issue by yipy0005** [iwatobipen/chemo_info#1]
Hi!

I was looking at the `HDBSCAN_Chemoinfo.ipynb` in the `chemicalspace2` folder and I saw these 2 blocks of code:

```
import numba

@numba.njit()
def tanimoto_dist(a,b):
    dotprod = np.dot(a,b)
    tc = dotprod / (np.sum(a) + np.sum(b) - dotprod)
    return 1.0-tc`

tsne = TSNE(n_components=2, metric=tanimoto_dist)
tsne_X = tsne.fit_transform(X)
print(tsne_X.shape)
```

I would like to ask if we actually can set the function as the metric or is there an intermediate step to generate the tanimoto distances to be used as the metric?

Thank you!

Best,
Yew Mun
